### PR TITLE
Consolidate Emit Array code and convert to sdk style csproj

### PR DESCRIPTION
**Pull request by pinzart90**
### Purpose

Moved the array emit code into dedicated function `EmitArray`. This should make it safer and easier to emit arrays
Converted the csproj to sdk style

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/Engine/EmitMSIL/CodeGenIL.cs [hunk 1]
```diff
@@ -528,50 +515,100 @@ private Type EmitExprListNode(AssociativeNode node)
...
-                else if (ot == typeof(long))
-                {
-                    EmitOpCode(OpCodes.Stelem_I8);
-                }
-                else if (ot == typeof(double))
+                if (ot == typeof(double))
                 {
                     if (t == typeof(int) || t == typeof(long))
                     {
                         EmitOpCode(OpCodes.Conv_R8);
```
Comment (pinzart90) [inline src/Engine/EmitMSIL/CodeGenIL.cs:529]: maybe we should add more conversion cases here ?
ex. what if `ot` is long and t is `double`

File: src/Engine/EmitMSIL/CodeGenIL.cs [hunk 2]
```diff
@@ -528,50 +515,100 @@ private Type EmitExprListNode(AssociativeNode node)
...
+            {
+                object obj = item;
+                if(t == typeof(int))
+                {
+                    EmitOpCode(OpCodes.Ldc_I4, (int)obj);
+                }
+                else if(t == typeof(long)) 
                 {
-                    EmitOpCode(OpCodes.Stelem_I1);
+                    EmitOpCode(OpCodes.Ldc_I8, (long)obj);
```
Comment (pinzart90) [inline src/Engine/EmitMSIL/CodeGenIL.cs:553]: this uses boxing/unboxing
Performance might be affected

File: src/Engine/EmitMSIL/CodeGenIL.cs [hunk 3]
```diff
@@ -528,50 +515,103 @@ private Type EmitExprListNode(AssociativeNode node)
...
+        /// <typeparam name="T"></typeparam>
+        /// <param name="arrType">The type of the array</param>
+        /// <param name="items">The list that will be iterated</param>
+        /// <param name="itemEmitter">A callback on each of the item's items</param>
+        private void EmitArray<T>(Type arrType, IEnumerable<T> items, Action<T, int> itemEmitter)
+        {
+            EmitOpCode(OpCodes.Ldc_I4, items == null ? 0 : items.Count());
+            EmitOpCode(OpCodes.Newarr, arrType);
+
+            if (items == null) return;
```
Comment (aparajit-pratap) [inline src/Engine/EmitMSIL/CodeGenIL.cs:589]: Shouldn't this check be the first statement in this function?

Comment (aparajit-pratap) [reply src/Engine/EmitMSIL/CodeGenIL.cs:589]: Or is there a useful case for creating an empty array? Maybe.

Comment (aparajit-pratap) [reply src/Engine/EmitMSIL/CodeGenIL.cs:589]: I see now. never mind the above question.

File: src/Engine/EmitMSIL/CodeGenIL.cs [hunk 4]
```diff
@@ -528,50 +515,103 @@ private Type EmitExprListNode(AssociativeNode node)
...
+                else if(t == typeof(char))
+                {
+                    EmitOpCode(OpCodes.Ldc_I4_S, (char)obj);
+                }
+                else if(t == typeof(string))
+                {
+                    EmitOpCode(OpCodes.Ldstr, (string)obj);
+                }
+            });
+        }
```
Comment (aparajit-pratap) [inline src/Engine/EmitMSIL/CodeGenIL.cs:574]: Do you think this function is really useful? Right now, it replaces only 2 lines of code, lines 670 and 671 before this PR.

**Review by aparajit-pratap — APPROVED** [review]
Refactor makes sense. Thanks!

File: src/Engine/EmitMSIL/CodeGenIL.cs (hunk 4 above)
Comment (pinzart90) [reply src/Engine/EmitMSIL/CodeGenIL.cs:574]: Maybe not useful right now.
I can remove it for now and lines 670 and 671 could be replaced with a call to the other EmitArray function: 
EmitArray<string>(typeof(string), items: null, itemEmitter: null);

**Review by aparajit-pratap — APPROVED** [review]

**Comment by mjkkirschner** [comment]
@pinzart90 sorry - I caused some conflicts on this PR.

**Comment by pinzart90** [comment]
2 performance tests fail
```
[Dynamo.Tests.PerformanceTests.PerformanceTest("C:\\Jenkins\\workspace\\Dynamo\\DynamoSelfServe\\pullRequestValidation\\Dynamo\\test\\core\\performance\\aniform.dyn")](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/6517/testReport/junit/Dynamo/Tests/PerformanceTests_PerformanceTest__C___Jenkins__workspace__Dynamo__DynamoSelfServe__pullRequestValidation__Dynamo__test__core__performance__aniform_dyn__/)
[Dynamo.Tests.PerformanceTests.PerformanceTest("C:\\Jenkins\\workspace\\Dynamo\\DynamoSelfServe\\pullRequestValidation\\Dynamo\\test\\core\\performance\\lotsofcoloredstuff.dyn")](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/6517/testReport/junit/Dynamo/Tests/PerformanceTests_PerformanceTest__C___Jenkins__workspace__Dynamo__DynamoSelfServe__pullRequestValidation__Dynamo__test__core__performance__lotsofcoloredstuff_dyn__/)
```
I think these are expected to fail for now.

**Pull request merged**